applications/graphs/views.py

```diff
@@ -762,763 +863,851 @@ def _get_graph_groups(request, graph_id, query={}):
+
+
+    Parameters
+    ----------
+    layout : dict
+            Dictionary containing the data of the layout being added.
+    layout_id : string
+            Unique ID of the layout.
+    request : object
+            HTTP POST Request.
+
+    Returns
+    -------
+    layout : object
+            Updated layout object.
+
+    Raises
+    ------
+
+    Notes
+    ------
+    It will update the owner_email only if user has admin access otherwise user cannot update the owner email.
+
+    """
+    authorization.validate(
+        request, permission='LAYOUT_UPDATE', layout_id=layout_id)
+    user_role = authorization.user_role(request)
+
+    return_value = utils.serializer(graphs.update_layout(request, layout_id,
+                                                         owner_email=layout.get('owner_email',
+                                                                                None) if user_role == authorization.UserRole.ADMIN else None,
+                                                         name=layout.get(
+                                                             'name', None),
+                                                         graph_id=layout.get(
+                                                             'graph_id', None),
+                                                         is_shared=layout.get(
+                                                             'is_shared', None),
+                                                         positions_json=layout.get(
+                                                             'positions_json', None),
+                                                         style_json=layout.get(
+                                                             'style_json', None),
+                                                         ))
+
+    # Notification
+    if layout.get('is_shared', None) == 1:
+        group_ids = [utils.serializer(group)['id'] for group in users.get_groups_by_graph_id(request, graph_id=graph_id)]
+        producer.send_message('group', {
+            'group_ids': group_ids,   
+            'message': settings.NOTIFICATION_MESSAGE['group']['share']['layout'].format(name=return_value.get('name','')),
+            'resource': 'layout',
+            'resource_id': layout_id,
+            'type': 'share',
+            'owner_email': request.session.get('uid', None)
+            })
+    elif layout.get('is_shared', None) == 0:
+        group_ids = [utils.serializer(group)['id'] for group in users.get_groups_by_graph_id(request, graph_id=graph_id)]
+        producer.send_message('group', {
+            'group_ids': group_ids,   
+            'message': settings.NOTIFICATION_MESSAGE['group']['unshare']['layout'].format(name=return_value.get('name','')),
+            'resource': 'layout',
+            'resource_id': layout_id,
+            'type': 'unshare',
+            'owner_email': request.session.get('uid', None)
+            })
+    else:
```

Comment:
Can you clarify this in the code with the comment, to make it explicit that this is by design and not a logic flaw at the moment?